src/graph/RadarPostprocessPointsNode.cpp

```diff
@@ -0,0 +1,195 @@
+// Copyright 2023 Robotec.AI
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <algorithm>
+
+#include <graph/NodesCore.hpp>
+#include <graph/NodesPcl.hpp>
+#include <gpu/nodeKernels.hpp>
+
+void RadarPostprocessPointsNode::setParameters(float distanceSeparation, float azimuthSeparation)
+{
+	this->distanceSeparation = distanceSeparation;
+	this->azimuthSeparation = azimuthSeparation;
+}
+
+void RadarPostprocessPointsNode::validateImpl()
+{
+	IPointsNodeSingleInput::validateImpl();
+
+	// Needed to clear cache because fields in the pipeline may have changed
+	// In fact, the cache manager is no longer useful here
+	// To be kept/removed in some future refactor (when resolving comment in the `enqueueExecImpl`)
+	cacheManager.clear();
+}
+
+void RadarPostprocessPointsNode::enqueueExecImpl()
+{
+	cacheManager.trigger();
+
+	if (input->getPointCount() == 0) {
+		filteredIndices->resize(0, false, false);
+		return;
+	}
+
+	distanceInputHost->copyFrom(input->getFieldData(DISTANCE_F32));
+	azimuthInputHost->copyFrom(input->getFieldData(AZIMUTH_F32));
+
+	std::vector<RadarCluster> clusters;
+	// Create first cluster with the first point
+	clusters.emplace_back(0, distanceInputHost->getReadPtr()[0], azimuthInputHost->getReadPtr()[0]);
+
+	for (int i = 1; i < input->getPointCount(); ++i) {
+		auto distance = distanceInputHost->getReadPtr()[i];
+		auto azimuth = azimuthInputHost->getReadPtr()[i];
+		bool isPointClustered = false;
+		for (auto&& cluster : clusters) {
+			if (cluster.isCandidate(distance, azimuth, distanceSeparation, azimuthSeparation)) {
+				cluster.addPoint(i, distance, azimuth);
+				isPointClustered = true;
+				break;
+			}
+		}
+
+		if (!isPointClustered) {
+			// Create a new cluster
+			clusters.emplace_back(i, distance, azimuth);
+		}
+	}
+
+	// Merge clusters if are close enough
+	bool allClustersGood = false;
+	while (clusters.size() < 2 || !allClustersGood) {
```

Comment:
```suggestion
	while (clusters.size() >= 2 && !allClustersGood) {
```
Was this maybe your intention? With above, it will break, when only one cluster left or all clusters are good.